src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IDocumentSnapshot.cs

```diff
@@ -16,6 +16,8 @@ internal interface IDocumentSnapshot
     IProjectSnapshot Project { get; }
     bool SupportsOutput { get; }
 
+    int Version { get; }
```

Comment:
It feels odd not to use `VersionStamp` here. Hoping that comes in a later commit. :smile:

Reply:
`VersionStamp` has no way to reason about ordering (ie, which version is newer than which) and our CustomMessageTarget code has to do that. In fact, it was a bit of a pain to add the equivalent logic to cohosting because Roslyn only exposed VersionStamp. `int` is much easier for this purpose.

Reply:
That makes sense. I still recommend renaming `_version` to `_textVersion` in `IDocumentSnapshot` and making `numericVersion` just `version`.